examples/containercopyinto.rs

```diff
@@ -18,10 +18,12 @@ fn main() {
     file.read_to_end(&mut bytes)
         .expect("Cannot read file on the localhost.");
 
-    let fut = docker
+    if let Err(e) = docker
```

Comment:
Instead of doing `if let` I would set the main to return a Result<(), shiplift::Error> and then just return `docker..` like :https://github.com/hyperium/hyper/blob/master/examples/echo.rs

Reply:
I think the only benefit of doing this in practice is to be able to do early returns using the `?` operator. Since we are handling and printing the errors, I don't think we gain anything by returning a result from main. You get an arbitrary exit code too I suppose (instead of '0').

Further, it looks like many of these examples are set to use environment variables (some specify the exact command in comments). This makes me think that this is used in some kind of external test rig (see my above question). For this reason I've tried to ensure that the output of the scripts is as similar as possible to the original behaviour.

Are there other reasons to return a result from main that are of benefit here?

Reply:
Nope, I like your point of keeping the behaviour as similar to the original as possible.